plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/unsegmented/RocksDBKeyValueStorage.java

```diff
@@ -112,39 +117,29 @@ public boolean containsKey(final byte[] key) throws StorageException {
         rocksDBMetrics.getReadLatency().startTimer()) {
       return Optional.ofNullable(db.get(key));
     } catch (final RocksDBException e) {
-      throw new StorageException(e);
+      throw RocksDBExceptionAdapter.createStorageException(e);
     }
   }
 
   @Override
-  public long removeAllKeysUnless(final Predicate<byte[]> retainCondition) throws StorageException {
-    long removedNodeCounter = 0;
-    try (final RocksIterator rocksIterator = db.newIterator()) {
-      for (rocksIterator.seekToFirst(); rocksIterator.isValid(); rocksIterator.next()) {
-        final byte[] key = rocksIterator.key();
-        if (!retainCondition.test(key)) {
-          removedNodeCounter++;
-          db.delete(key);
-        }
-      }
-    } catch (final RocksDBException e) {
-      throw new StorageException(e);
-    }
-    return removedNodeCounter;
+  public Set<byte[]> getAllKeysThat(final Predicate<byte[]> returnCondition) {
+    return streamKeys().filter(returnCondition).collect(toUnmodifiableSet());
```

Comment:
We were previously using `Sets.newIdentityHashSet()` to create this set, which it looks like we were doing in order to make the equality checks more efficient.  We should probably keep that.

Reply:
I thought that was already the case for `byte[]`.
```
 ~/wrk/besu pruning-bugfix-doomed-key ⁘ jshell
|  Welcome to JShell -- Version 11.0.6
|  For an introduction type: /help intro

jshell> new HashSet<byte[]>()
$1 ==> []

jshell> $1.add(new byte[] {1})
$2 ==> true

jshell> $1
$1 ==> [[B@5034c75a]

jshell> $1.contains(new byte[] {1})
$4 ==> false

jshell>
```

Reply:
Yeah - I think you're right.  Did some quick googling trying to figure out why we were using `newIdentityHashSet` and I misread an answer that referenced: https://docs.oracle.com/javase/6/docs/api/java/util/Arrays.html#equals%28byte%5B%5D,%20byte%5B%5D%29